src/runtime/vector.h

```diff
@@ -2,22 +2,19 @@ typedef buffer vector;
 
 static inline void *vector_get(vector v, int offset)
 {
-    void *res;
     bytes base = v->start + offset * sizeof(void *);
```

Comment:
Minor nit involving existing code, but this function could be simpler. A vector is really a buffer, so you could perhaps just change offset to a byte offset, compare it against buffer_length, and if valid return (void **)buffer_ref(v, offset).